spec/controllers/projects_controller_spec.rb

```diff
@@ -1,13 +1,85 @@
 require 'spec_helper'
 
+RSpec.shared_examples 'denies access' do
```

Comment:
Don't worry about duplication in tests 😄 if duplication makes it easier to read then its fine.